### DISCUSSION TITLE
Is there a way to check if a Model has all attributes as None?

### DISCUSSION THREAD

**Discussion by celsoclemente** [pydantic/pydantic#4613] [Question] [answered]
I was thinking of a boolean method like: `user.empty` to quickly see if all of an objects attributes are None.

Do we have something like that?

**Comment by samuelcolvin** [comment, marked as answer]
Should be as simple as `bool(m.__fields_set__)`, though that's not quite the same as `None` since values not in `__fields_set__` will have their default value.

**Reply by samuelcolvin** [reply]
otherwise you'll probably just want `any(m.dict().values())`

**Comment by japita-se** [comment]
In pydantic v2 do:
`bool(m.model_fields_set)`
